api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/PersonAttributeTranslatorImpl.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.annotation.Nonnull;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+import org.hl7.fhir.r4.model.BooleanType;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.StringType;
+import org.hl7.fhir.r4.model.Type;
+import org.hl7.fhir.r4.model.UriType;
+import org.openmrs.Concept;
+import org.openmrs.Location;
+import org.openmrs.PersonAttribute;
+import org.openmrs.PersonAttributeType;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.PersonService;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.translators.ConceptTranslator;
+import org.openmrs.module.fhir2.api.translators.PersonAttributeTranslator;
+import org.openmrs.module.fhir2.api.util.FhirUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class PersonAttributeTranslatorImpl implements PersonAttributeTranslator {
+	
+	private static final String JAVA_STRING_FORMAT = "java.lang.String";
+	
+	private static final String JAVA_BOOLEAN_FORMAT = "java.lang.Boolean";
+	
+	private static final String OPENMRS_LOCATION_FORMAT = "org.openmrs.Location";
+	
+	private static final String OPENMRS_CONCEPT_FORMAT = "org.openmrs.Concept";
+	
+	private static final String LOCATION_TYPE = "Location";
+	
+	@Autowired
+	private LocationService locationService;
+	
+	@Autowired
+	private PersonService personService;
```

Comment:
We have a pretty strict rule about not using core OpenMRS services in the translation layer, which I think it's preferable to follow.